hoot-core/src/main/cpp/hoot/core/io/OsmApiChangeset.cpp

```diff
@@ -189,6 +188,98 @@ void XmlChangeset::loadElements(QXmlStreamReader& reader, ChangesetType changese
   }
 }
 
+void XmlChangeset::splitLongWays(long maxWayNodes)
+{
+  //  Negative numbers means there is no max set
+  if (maxWayNodes < 0)
+    return;
+  //  Iterate all of the changeset types except delete
+  for (int type = TypeCreate; type < TypeDelete; ++type)
+  {
+    for (XmlElementMap::iterator it = _ways[type].begin(); it != _ways[type].end(); ++it)
+    {
+      XmlWay* way = dynamic_cast<XmlWay*>(it->second.get());
+      while (way->getNodeCount() > maxWayNodes)
+      {
+        //  Create a copy of the way
+        XmlWayPtr newWay(new XmlWay(*way));
+        newWay->changeId(getNextWayId());
+        //  Remove maxWayNodes from the original and add them to this way
+        way->removeNodes(0, maxWayNodes - 1);
+        newWay->removeNodes(maxWayNodes);
+        //  Save the new way
+        _idMap.addId(ElementType::Way, newWay->id());
+        _ways[TypeCreate][newWay->id()] = newWay;
+        _allWays[newWay->id()] = newWay;
+      }
+    }
+  }
+}
```

Comment:
Function to loop through each way and make sure that it doesn't exceed the max number of nodes per way from the API's capabilities call.  New ways are created for the splits.